### PR TITLE
Jwe builder fat  part 2

### PR DESCRIPTION
**Pull request by c00crane**
Deliver the renaming JWE Builder FAT

### CODE REVIEW

**Comment by c00crane** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.jwt_fat.builder,com.ibm.ws.security.jwt_fat.consumer,com.ibm.ws.security.jwtsso_fat,com.ibm.ws.security.mp.jwt.1.1_fat,com.ibm.ws.security.mp.jwt.1.2_fat
#run-libby-bot

**Comment by c00crane** [comment]
#13576

File: dev/com.ibm.ws.security.jwt_fat.builder/fat/src/com/ibm/ws/security/jwt/fat/builder/JwtBuilderApiBasicTests.java [hunk 1]
```diff
@@ -5353,745 +5597,1022 @@ public void JwtBuilderAPIBasicTests_multiple_apis_and_consumeToken() throws Exce
...
+        String encryptKey = JwtKeyTools.getComplexPublicKeyForSigAlg(builderServer, JWTBuilderConstants.SIGALG_RS256);
+        String decryptKey = JwtKeyTools.getComplexPrivateKeyForSigAlg(builderServer, JWTBuilderConstants.SIGALG_RS256);
+
+        testSettings.put(JWTBuilderConstants.ENCRYPT_KEY, encryptKey);
+        testSettings.put(JWTBuilderConstants.DECRYPT_KEY, decryptKey);
+        testSettings.put(JWTBuilderConstants.CONTENT_ENCRYPT_ALG, JWTBuilderConstants.DEFAULT_CONTENT_ENCRYPT_ALG);
+
+        Expectations expectations = BuilderHelpers.createBadBuilderExpectations(JWTBuilderConstants.JWT_BUILDER_SETAPIS_ENDPOINT, JwtBuilderMessageConstants.CWWKS6060E_CAN_NOT_CREATE_JWE, builderServer);
+        expectations.addExpectation(new ServerMessageExpectation(builderServer, JwtBuilderMessageConstants.CWWKS6020E_CAN_NOT_CAST, "Message log did not contain an error indicating a problem trying to encrypt the token."));
+        expectations.addExpectation(new ServerMessageExpectation(builderServer, JwtBuilderMessageConstants.CWWKS6060E_CAN_NOT_CREATE_JWE, "Message log did not contain an error indicating that the key was not large enough."));
```
Comment (ayoho) [inline dev/com.ibm.ws.security.jwt_fat.builder/fat/src/com/ibm/ws/security/jwt/fat/builder/JwtBuilderApiBasicTests.java:6255]: Not terribly important, but the failure message here references the key not being large enough but the actual failure will be for an unknown/invalid key management algorithm.

File: dev/com.ibm.ws.security.jwt_fat.builder/fat/src/com/ibm/ws/security/jwt/fat/builder/JwtBuilderApiBasicTests.java [hunk 2]
```diff
@@ -5353,745 +5597,1022 @@ public void JwtBuilderAPIBasicTests_multiple_apis_and_consumeToken() throws Exce
...
+        String encryptKey = JwtKeyTools.getComplexPublicKeyForSigAlg(builderServer, JWTBuilderConstants.SIGALG_RS256);
+        String decryptKey = JwtKeyTools.getComplexPrivateKeyForSigAlg(builderServer, JWTBuilderConstants.SIGALG_RS256);
+
+        testSettings.put(JWTBuilderConstants.ENCRYPT_KEY, encryptKey);
+        testSettings.put(JWTBuilderConstants.DECRYPT_KEY, decryptKey);
+        testSettings.put(JWTBuilderConstants.CONTENT_ENCRYPT_ALG, "someContentEncryptAlg");
+
+        Expectations expectations = BuilderHelpers.createBadBuilderExpectations(JWTBuilderConstants.JWT_BUILDER_SETAPIS_ENDPOINT, JwtBuilderMessageConstants.CWWKS6060E_CAN_NOT_CREATE_JWE, builderServer);
+        expectations.addExpectation(new ServerMessageExpectation(builderServer, JwtBuilderMessageConstants.CWWKS6020E_CAN_NOT_CAST, "Message log did not contain an error indicating a problem trying to encrypt the token."));
+        expectations.addExpectation(new ServerMessageExpectation(builderServer, JwtBuilderMessageConstants.CWWKS6060E_CAN_NOT_CREATE_JWE, "Message log did not contain an error indicating that the key was not large enough."));
```
Comment (ayoho) [inline dev/com.ibm.ws.security.jwt_fat.builder/fat/src/com/ibm/ws/security/jwt/fat/builder/JwtBuilderApiBasicTests.java:6362]: Same comment - need different failure string.

File: dev/com.ibm.ws.security.jwt_fat.builder/fat/src/com/ibm/ws/security/jwt/fat/builder/JwtBuilderApiBasicTests.java [hunk 3]
```diff
@@ -5353,745 +5597,1022 @@ public void JwtBuilderAPIBasicTests_multiple_apis_and_consumeToken() throws Exce
...
         validationUtils.validateResult(response, expectations);
 
+        validationUtils.validateJWEToken(response, (String) testSettings.get(JWTBuilderConstants.KEY_MGMT_ALG), (String) testSettings.get(JWTBuilderConstants.DECRYPT_KEY), (String) testSettings.get(JWTBuilderConstants.CONTENT_ENCRYPT_ALG));
+
     }
 
+    @SkipForRepeat(JwtBuilderClaimRepeatActions.SingleID)
     @Test
-    public void JwtBuilderAPIBasicTests_claimsFrom_jweString_decodedPayload_allClaims() throws Exception {
+    public void JwtBuilderAPIBasicTests_encryptWith_useDifferentPublicKeyThanBuilderConfig() throws Exception {
```
Comment (ayoho) [inline dev/com.ibm.ws.security.jwt_fat.builder/fat/src/com/ibm/ws/security/jwt/fat/builder/JwtBuilderApiBasicTests.java:6566]: I'm not seeing any difference between this test and `JwtBuilderAPIBasicTests_encryptWith_useSameSettingsAsBuilderConfig`.

**Review by ayoho — APPROVED** [review]

File: dev/com.ibm.ws.security.jwt_fat.builder/fat/src/com/ibm/ws/security/jwt/fat/builder/JwtBuilderApiBasicTests.java (hunk 3 above)
Comment (c00crane) [reply dev/com.ibm.ws.security.jwt_fat.builder/fat/src/com/ibm/ws/security/jwt/fat/builder/JwtBuilderApiBasicTests.java:6566]: Cut/paste error:  JwtBuilderAPIBasicTests_encryptWith_useSameSettingsAsBuilderConfig should have used RS256 to override the RS256 in the config.  Thought I'd fixed that.
Will deliver with the consumer tests I'm working now.

File: dev/com.ibm.ws.security.jwt_fat.builder/fat/src/com/ibm/ws/security/jwt/fat/builder/JwtBuilderApiBasicTests.java (hunk 2 above)
Comment (c00crane) [reply dev/com.ibm.ws.security.jwt_fat.builder/fat/src/com/ibm/ws/security/jwt/fat/builder/JwtBuilderApiBasicTests.java:6362]: Updated msg to log the correct reason for failure.

File: dev/com.ibm.ws.security.jwt_fat.builder/fat/src/com/ibm/ws/security/jwt/fat/builder/JwtBuilderApiBasicTests.java (hunk 1 above)
Comment (c00crane) [reply dev/com.ibm.ws.security.jwt_fat.builder/fat/src/com/ibm/ws/security/jwt/fat/builder/JwtBuilderApiBasicTests.java:6255]: Updated msg to log the correct reason for failure.

**Pull request merged**